### PR TITLE
fix: noop update from selectors should work

### PR DESCRIPTION
**Pull request by Weakky**
## Overview

Follow-up to https://github.com/prisma/prisma-engines/pull/4084. Fixes a regression for 1-1 update when relationMode=prisma.

The PR fixes two things:
1. Noop updates take selectors into account: In the case of an update with `relationMode=prisma`, we read the record to update _before_ doing the actual update to compute whether some referential actions will need to be added to the graph. When doing this additional read, we pass the read ids down to the update operation so that we save reading the ids again when the connector does not support `UpdateReturning`. These ids are passed as `RecordFilter.selectors`. However, the selectors were not considered when doing a simple read for a noop update. Therefore, the update node wasn't returning the correct ids, leading to failed operations later on.
2. `WriteArgs::returns` is now updated to take into account the fact that update nodes can now return more than just the primary identifier of a model. While we haven't noticed any bug, it could've resulted in missing reload nodes when the child of a node is requesting data dependencies that aren't fulfilled by the parent's selection set.

See https://prisma-company.slack.com/archives/C058VM009HT/p1690562457715389

### CODE REVIEW

File: query-engine/core/src/query_ast/write.rs [hunk 1]
```diff
@@ -64,7 +64,13 @@ impl WriteQuery {
...
             Self::CreateRecord(_) => returns_id,
             Self::CreateManyRecords(_) => false,
-            Self::UpdateRecord(_) => returns_id,
+            Self::UpdateRecord(UpdateRecord::WithExplicitSelection(ur)) => {
+                ur.selected_fields.is_superset_of(field_selection)
+            }
+            Self::UpdateRecord(UpdateRecord::WithImplicitSelection(ur)) => {
+                ur.selected_fields().is_superset_of(field_selection)
+            }
+            Self::UpdateRecord(UpdateRecord::WithoutSelection(_)) => returns_id,
```
Comment (Weakky) [inline query-engine/core/src/query_ast/write.rs:73]: This is a bit out of scope for this PR.

This method, `WriteQuery::returns`, is used to compute reload nodes by checking whether a node's field selection fulfills its children data dependencies. It had not been updated after making `UpdateRecord` nodes return an arbitrary selection set. This means that some reload nodes might've not been added in cases it's needed.

**Review by miguelff — APPROVED** [review]
